test_matrix.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from matrix import Matrix
+
+
+class TestMatrix(unittest.TestCase):
+    """Tests my matrix methods."""
+
+    def setUp(self):
+        self.a = Matrix(((1, 2), (3, 4)))
+        self.b = Matrix(((5, 6), (7, 8)))
+        self.c = Matrix(((1, 2, 3), (4, 5, 6), (7, 8, 9)))
+        self.d = Matrix(((2, 3, 4), (5, 6, 7), (7, 8, 9)))
+
+    def tearDown(self):
+        pass
+
+    def test_init(self):
+        with self.assertRaises(ValueError):
+            self.e = Matrix(((1, 2), (1, 2, 3)))  # not nXn
+
+    def test_add(self):
+        self.assertEqual(Matrix.__add__(self.a, self.b),
+                         Matrix(((6, 8), (10, 12))))
+        self.assertEqual(Matrix.__add__(self.c, self.d),
+                         Matrix(((3, 5, 7), (9, 11, 13), (14, 16, 18))))
+        # test addition of matrices with different dimensions
+        self.assertEqual(Matrix.__add__(self.a, self.c), None)
```

Comment:
You compare between two matrix objects, which means the _ _ eq _ _ function is used to do it,
This mean instead of checking a single functionality here you check 2.
This mean that you can have a bug in add, but the test won't fail because you also have bug in eq function (for example it always return true)